src/main/java/emu/grasscutter/server/packet/recv/HandlerUnlockTransPointReq.java

```diff
@@ -0,0 +1,16 @@
+package emu.grasscutter.server.packet.recv;
+
+import emu.grasscutter.net.packet.Opcodes;
+import emu.grasscutter.net.packet.PacketOpcodes;
+import emu.grasscutter.net.proto.UnlockTransPointReqOuterClass.UnlockTransPointReq;
+import emu.grasscutter.net.packet.PacketHandler;
+import emu.grasscutter.server.game.GameSession;
+
+@Opcodes(PacketOpcodes.UnlockTransPointReq)
+public class HandlerUnlockTransPointReq extends PacketHandler {
+    @Override
+    public void handle(GameSession session, byte[] header, byte[] payload) throws Exception {
+        UnlockTransPointReq req = UnlockTransPointReq.parseFrom(payload);
+        session.getPlayer().getProgressManager().unlockTransPoint(req.getSceneId(), req.getPointId());
+    }
```

Comment:
I also got PlayerPropChangeReasonNotify, PlayerPropNotify, and ItemAddHintNotify for adventure exp. official server sent me 10, but you get 50 adventure exp, so maybe the exp modifier should be changed? There's also the handbook investigation target, but that is not implemented yet
```suggestion
    }
    session.getPlayer().getQuestManager().triggerEvent(QuestTrigger.QUEST_CONTENT_UNLOCK_TRANS_POINT, req.getSceneId(), req.getPointId());

        // Give Adventure Exp.
        session.getPlayer().getInventory().addItem(102,10, ActionReason.getTypeByValue(19));
        session.getPlayer().earnExp(10);

        session.send(new PacketPlayerPropChangeReasonNotify(session.getPlayer().getProperty(PROP_PLAYER_EXP),
            PROP_PLAYER_EXP, PropChangeReasonOuterClass.PropChangeReason.PROP_CHANGE_REASON_PLAYER_ADD_EXP));
```

Reply:
Oh, I completely forgot that unlocking waypoints gives Adventure EXP. I'll add that tomorrow.